### PR TITLE
Fix WeightedSelector multiple option select

### PR DESCRIPTION
**Pull request by ipdae**
- Fix `WeightedSelector.Select` when multiple options select

### CODE REVIEW

**Review by boscohyun — APPROVED** [review]

**Review by Unengine — APPROVED** [review]

**Review by hskim881028 — APPROVED** [review]

File: Lib9c/Battle/Simulator.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ MaterialItemSheet materialItemSheet
             {
                 try
                 {
-                    var data = itemSelector.Select(1).First();
+                    var data = itemSelector.SelectV2(1).First();
```
Comment (riemannulus) [inline Lib9c/Battle/Simulator.cs:71]: It is ok for block backwards compatibility before?

File: Lib9c/Action/CombinationEquipment3.cs [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+            {
+                throw new CombinationSlotUnlockException(
+                    $"Aborted as the slot state is invalid: {slotState} @ {SlotIndex}");
+            }
+
+            var recipeSheet = states.GetSheet<EquipmentItemRecipeSheet>();
+            var materialSheet = states.GetSheet<MaterialItemSheet>();
+            var materials = new Dictionary<Material, int>();
+
+            // 레시피 검증
```
Comment (riemannulus) [inline Lib9c/Action/CombinationEquipment3.cs:74]: Can we translate to English this comment?

File: Lib9c/Action/CombinationEquipment3.cs [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+            {
+                if (!recipe.SubRecipeIds.Contains((int) SubRecipeId))
+                {
+                    throw new SheetRowColumnException(
+                        $"Aborted as the sub recipe {SubRecipeId} was failed to load from the sheet."
+                    );
+                }
+            }
+
+            // 메인 레시피 해금 검사.
```
Comment (riemannulus) [inline Lib9c/Action/CombinationEquipment3.cs:90]: Can we translate to English this comment?

File: Lib9c/Action/CombinationEquipment3.cs [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+            }
+
+            var equipmentMaterial = ItemFactory.CreateMaterial(materialSheet, material.Id);
+            materials[equipmentMaterial] = recipe.MaterialCount;
+
+            BigInteger requiredGold = recipe.RequiredGold;
+            var requiredActionPoint = recipe.RequiredActionPoint;
+            var equipmentItemSheet = states.GetSheet<EquipmentItemSheet>();
+
+            // 장비 제작
```
Comment (riemannulus) [inline Lib9c/Action/CombinationEquipment3.cs:116]: Can we translate to English this comment?

File: Lib9c/Action/CombinationEquipment3.cs [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+            }
+
+            var requiredBlockIndex = ctx.BlockIndex + recipe.RequiredBlockIndex;
+            var equipment = (Equipment) ItemFactory.CreateItemUsable(
+                equipRow,
+                ctx.Random.GenerateRandomGuid(),
+                requiredBlockIndex
+            );
+
+            // 서브 레시피 검증
```
Comment (riemannulus) [inline Lib9c/Action/CombinationEquipment3.cs:129]: Can we translate to English this comment?

File: Lib9c/Action/CombinationEquipment3.cs [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+                    var subMaterial = ItemFactory.CreateMaterial(materialSheet, materialInfo.Id);
+                    materials[subMaterial] = materialInfo.Count;
+                }
+
+                optionIds = SelectOption(states.GetSheet<EquipmentItemOptionSheet>(), states.GetSheet<SkillSheet>(),
+                    subRecipe, ctx.Random, equipment);
+                equipment.Update(requiredBlockIndex);
+            }
+
+            // 자원 검증
```
Comment (riemannulus) [inline Lib9c/Action/CombinationEquipment3.cs:168]: Can we translate to English this comment?

File: Lib9c/Action/CombinationEquipment3.cs [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+            avatarState.actionPoint -= requiredActionPoint;
+            if (!(optionIds is null))
+            {
+                foreach (var id in optionIds.OrderBy(id => id))
+                {
+                    agentState.unlockedOptions.Add(id);
+                }
+            }
+
+            // FIXME: BlacksmithAddress 계좌로 돈이 쌓이기만 하는데 이걸 어떻게 순환시킬지 기획이 필요.
```
Comment (riemannulus) [inline Lib9c/Action/CombinationEquipment3.cs:195]: Can we translate to English this comment?

File: Lib9c/Battle/Simulator.cs (hunk 1 above)
Comment (ipdae) [reply Lib9c/Battle/Simulator.cs:71]: yes. check this case in `WeightedSelectorTest.SelectV2Single`

**Pull request merged**